### PR TITLE
Github collector : retry max 5 for error 

### PR DESCRIPTION
**Pull request by nireeshT**
**Affects:** \<github-graphql-collector-version-number>.

---

### CODE REVIEW

**Review by nameisaravind — APPROVED** [review]
lgtm

File: src/main/java/com/capitalone/dashboard/collector/DefaultGitHubClient.java [hunk 1]
```diff
@@ -1255,7 +1255,21 @@ private JSONObject getDataFromRestCallPost(GitHubParsed gitHubParsed, GitHubRepo
...
             graphqlUrl = settings.getGraphqlUrl();
         }
-        ResponseEntity<String> response = makeRestCallPost(graphqlUrl, repo.getUserId(), password, personalAccessToken, query);
+        ResponseEntity<String> response;
+        int retryCount = 0;
+        while(true) {
+            try {
+                response = makeRestCallPost(graphqlUrl, repo.getUserId(), password, personalAccessToken, query);
+                break;
+            } catch (Exception e) {
```
Comment (rvema) [inline src/main/java/com/capitalone/dashboard/collector/DefaultGitHubClient.java:1264]: This section of the code needs to better thought and implemented.

**Review by rvema — CHANGES_REQUESTED** [review]
Please address the review comments

File: src/main/java/com/capitalone/dashboard/collector/DefaultGitHubClient.java (hunk 1 above)
Comment (nireeshT) [reply src/main/java/com/capitalone/dashboard/collector/DefaultGitHubClient.java]: @rvema  added check to retry only if HTTPStatus code 502

**Review by rvema — APPROVED** [review]
lgtm

**Pull request merged**